### DISCUSSION TITLE
call_later doesn't wait to screen refresh

### DISCUSSION THREAD

**Discussion by climbus** [Textualize/textual#144] [General]
From doc string call_later method: "Run a callback after processing all messages and refreshing the screen."

If I have good understanding, I can deffer call methods after rendering app. 

I checked it in expample/sample.py and added sleep to function get_markdown:

```python
async def get_markdown(filename: str) -> None:                                
   time.sleep(1)                                                             
   with open(filename, "rt") as fh:                                          
        readme = Markdown(fh.read(), hyperlinks=True)                         
   await body.update(readme)
```

I see black screen for 1 second after start. What am I doing wrong? 

I need to defer loading content after app first screen renders.

Thanks

**Comment by willmcgugan** [comment]
Using the non-async sleep is probably not a fair comparison since that will block everything, but there is still a flaw there. The `call_later` should be called after a refresh. I'll have a fix for that in a later version.

**Reply by climbus** [reply]
Ok, thanks
